### PR TITLE
RESOURCE-148 Support azure power bi capacity workloads

### LINKED ISSUE #466: Power BI Capacity Workloads

### PR DESCRIPTION
**Pull request by sathish-progress**
### Description
Support azure power bi capacity workload(s)

### Issues Resolved
#466 

### Check List

- [X] New functionality includes tests
- [X] All tests pass
- [X] `rake lint` passes
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/azure_power_bi_capacity_workload.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require 'azure_generic_resource'
+
+class AzurePowerBICapacityWorkload < AzureGenericResource
+  name 'azure_power_bi_capacity_workload'
+  desc 'Retrieves and verifies the settings of an Azure Power BI Capacity Workload.'
+  example <<-EXAMPLE
+    describe azure_power_bi_capacity_workload(capacity_id: '0f084df7-c13d-451b-af5f-ed0c466403b2', name: 'cfafbeb1-8037-4d0c-896e-a46fb27ff229') do
```
Comment (soumyo13) [inline libraries/azure_power_bi_capacity_workload.rb:7]:
```suggestion
    describe azure_power_bi_capacity_workload(capacity_id: 'CAPACITY_ID', name: 'WORKLOAD_NAME') do
```

**Review by soumyo13 — CHANGES_REQUESTED** [review]
Terraform part is not yet updated.

File: libraries/azure_power_bi_capacity_workload.rb (hunk 1 above)
Comment (sathish-progress) [reply libraries/azure_power_bi_capacity_workload.rb:7]: no here it has be original values in example, only in docs we specify that notation!

File: libraries/azure_power_bi_capacity_workload.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      it { should exist }
+    end
+  EXAMPLE
+
+  AUDIENCE = 'https://analysis.windows.net/powerbi/api'.freeze
+
+  def initialize(opts = {})
+    raise ArgumentError, 'Parameters must be provided in an Hash object.' unless opts.is_a?(Hash)
+
+    Validators.validate_parameters(resource_name: @__resource_name__, required: %i(capacity_id),
```
[the remark this answers is not in the document]
Comment (sathish-progress) [reply libraries/azure_power_bi_capacity_workload.rb:17]: handled implicitly here https://github.com/inspec/inspec-azure/blob/main/libraries/azure_backend.rb#L185

File: libraries/azure_power_bi_capacity_workload.rb (hunk 1 above)
Comment (soumyo13) [reply libraries/azure_power_bi_capacity_workload.rb:7]: Thanks for the information.

**Review by soumyo13 — APPROVED** [review]

**Review by IanMadd — APPROVED** [review]

**Pull request merged**